### ISSUE TITLE
Rename "collect images" to "filter images"

### ISSUE THREAD

**Issue by elstoc** [darktable-org/darktable#8893]
"Collect" is not a particularly great name to describe this module. What it does is "filter your image collection", so "filter images" is a much more descriptive name. 

"recently used collections" should be similarly renamed "recently used filters", and the word "collection" should only be used when describing the entire set of images currently loaded into the library database (though TBH "library" would maybe be better for that).

**Comment by phweyland** [comment]
I'm not comfortable with this.
It's not that `filter` is wrong, but I find collection more appropriate. If filter can be ok to find images related to something (time, keyword, ..._) collection is better when I think folder, trip, subject or any kind of classification I've built.
Then when one enters a text in the text box, that doesn't filter the images, but the selected item (folders containing `xxx` for example).
I think also this `filter` feature is not a very effective one. Let's find images having a given text `xxx` in filename, tag or metadata at once ...
Last there is another filter feature (stars) called `view`, effective filter but limited ...

**Comment by elstoc** [comment]
Searching for some other ideas/interpretations, I took a look through the translations (the ones I could read anyway). The French translation of "collect images" already includes the word "filter", the dutch translation uses a word that literally translates as "catalog".

Translating the French one back to English (sort of), "filter collection" doesn't sound that bad.

Anyway, just some thoughts.

**Comment by jenshannoschwalm** [comment]
"Selection" ? Would be the result of a filter. At least in other common usage.

**Comment by quovadit** [comment]
I agree that 'collect' means to get something (like importing), so I would also prefer to call it 'filter'.

If it's called 'collect images' now, why can't it be 'filter images' then?
The goal is to show specific images.

**Comment by parafin** [comment]
More technical term could be "create a view". Probably too technical.

**Comment by elstoc** [comment]
Problem is that we need both a verb to describe what the collect module does and a related noun to describe the result. If it didn't already have a specific meaning in darktable, I'd advocate "group". An alternative might be "set" or "image set", though I'm not particularly happy with either.

**Comment by phweyland** [comment]
> If I had stamps from multiple countries, grouping them by country would be "organising" my collection,

I can agree with that. But I can also organize my main collection into different specific collections (cities, trees, birds...).
BTW collect module does  what Lightroom "dynamic collection" does.  

I'm not particularly in love with the collect module and the word "collect" may not be very appealing but it describes correctly what it is done (in the meaning "set a collection of images").

Filtering is not very different, I agree and we are playing with words here.

But I expect far more interactivity from a "filter", like the stars filter.
I expect also to find easily the images where the text "xxx" can be found, no matter the type of information, folder, filename, tag, geotag, title, description,... In theory I can do it with collect module, but it is painful.

**Comment by elstoc** [comment]
> specific collections

> dynamic collection

See both of these are modifying the word "collection" with another word, which is fine by me. "filtered collection" seems to me a perfectly ok synonym for both of those. But "collection" on its own seems wrong.

> In theory I can do it with collect module, but it is painful

I think you're arguing there for a revamp of the collect module, which I'm entirely in favour of.

**Comment by phweyland** [comment]
> > dynamic collection
> 
> See both of these are modifying the word "collection" with another word, which is fine by me.

Not really related. In Lightroom, "dynamic collection" is to be compared to "collection", which is static, concept that dt doesn't have.

> > In theory I can do it with collect module, but it is painful
> 
> I think you're arguing there for a revamp of the collect module, which I'm entirely in favour of.

Not necessarily. Finding the images with "xxx" across all information is a filtering feature, not a collect module one. 

I understand criticism about the "collect" name but it is not misleading, it describes what the module does.
Of course the collect module is based on filters. But the collect module is about organization, collection and set of images (those I want to work on). These collections are updated when images data change.

A filter can be seen as an interactive tool to find the needle in a haystack, when I need it. It gives me an instant result (which can be volatile) inside the collection I work on (still like stars filter). 
I would be happy with different tools as I won't loose my working set when filtering it. But that's another discussion of course.

To come back to the main subject, the collect module collects a set of images from my library, which sounds fine to me. Could be "collections" as well.

**Comment by elstoc** [comment]
"create collection" might be a better name for the module.

**Comment by quovadit** [comment]
I don't understand why filter is the wrong name for the collect module.
'Finding images with xxx across all information' is no filtering but a search!

**Comment by Solarer** [comment]
As a new user of darktable that had a few years of experience with Lightroom, I can confirm that this module was more of a hurdle to me than it should have been. 
"create collection" or "build collection" probably comes closer to what the module does and makes more sense when you think about the next module "recently used  collections".
-> First you create/build a collection, then you can use or reuse it.

Apart from the name, the second problem is that the default rule is "film roll" which is not very intuitive. The default should probably be "folder"

**Comment by elstoc** [comment]
In that vein, "generate collection" is probably the best I can think of. "Build" and "create" imply a permanence that "generate" does not and, as previously suggested, it firms up the idea of what a collection is.

> Apart from the name, the second problem is that the default rule is "film roll" which is not very intuitive.

That's a separate issue, and "film roll" basically is synonymous with "folder" - it just doesn't let you browse with a folder tree.

**Comment by elstoc** [comment]
Generally, it would be nice to reinforce the terms library, collection and selection in the UI wherever possible so that people have a better idea of what they are working with. For example, "import in-place" could become "add to library" and "remove" could become "remove from library".
